src/software/amazon/ionschema/Type.kt

```diff
@@ -0,0 +1,43 @@
+package software.amazon.ionschema
+
+import software.amazon.ion.IonValue
+import software.amazon.ionschema.internal.ConstraintInternal
+
+/**
+ * A Type consists of an optional name and zero or more constraints.
+ *
+ * Unless otherwise specified, the constraint `type: any` is automatically applied.
+ */
+interface Type {
+    /**
+     * Returns the name of the type;  if the type has no name, a string representing
+     * the definition of the type is returned.
+     */
+    fun name(): String
+
+    /**
+     * If the specified value violates one or more of this type's constraints,
+     * returns `false`, otherwise `true`.
+     */
+    fun isValid(value: IonValue) = validate(this, value, true).isValid()
```

Comment:
I think it could also be useful to have `IonReader` variants of `isValid` and `validate` so that users that don't currently depend on the DOM don't need to start. Internally the implementation may still operate on DOM objects retrieved from the given reader, but it wouldn't have to.

Reply:
Opened #91 to track this idea